libbeat/outputs/codec/json/json.go

```diff
@@ -58,6 +58,8 @@ func (e *Encoder) reset() {
 		gotype.Folders(
 			codec.MakeTimestampEncoder(),
 			codec.MakeBCTimestampEncoder(),
+			codec.MakeNanoTimestampEncoder(),
+			codec.MakeBCNanoTimestampEncoder(),
```

Comment:
choose one, but not both. MakeTimestampEncoder and MakeNanoTimestampEncoder have the same function signature. So to not confuse setup, select only one. Note to myself: add panic if encoders with same signature are passed.

Reply:
@urso you mean that remove `codec.MakeTimestampEncoder()` and `codec.MakeBCTimestampEncoder()` function from `gotype.Folders`, is it?

>  Note to myself: add panic if encoders with same signature are passed.

I don't know what mean. `MakeNanoTimestampEncoder` has add panic, you can check https://github.com/elastic/beats/pull/7576/files/8996d4f0e27415eb7d390ccc0da694bad76969bf#diff-1983a46731b9d0b0b4bc88ec99b97794R41. Or other changes?

Reply:
Yes, remove them.

The encoder uses reflection to choose the right function depending on the function paramters and actual type. One type must have only one function registered. e.g. `MakeTimestampEncoder` and `MakeNanoTimestampEncoder` have the same type `func(*time.Time, structform.ExtVisitor) error`. Which one shall the encoder use?

> MakeNanoTimestampEncoder has add panic, you can check elastic/beats/pull/7576/files/8996d4f0e27415eb7d390ccc0da694bad76969bf#diff-1983a46731b9d0b0b4bc88ec99b97794R41. Or other changes?

Stack-trace with content of actual error message would be helpful. I guess you mean the panic in the error right after `dtfmt.NewFormatter`?